syncopy/tests/test_mne_conv.py

```diff
@@ -0,0 +1,158 @@
+
+        This uses raw data, i.e., data without trial definition.
+        """
+
+        adata = self.adata
+        # Convert to MNE RawArray
+        ar = spy.io.mne_conv.raw_adata_to_mne(adata)
+        # Now convert back to AnalogData
+        adata2 = spy.io.mne_conv.raw_mne_to_adata(ar)
+        assert type(adata2) == spy.AnalogData
+        assert all(adata.channel == adata2.channel)
+        assert np.allclose(adata.data, adata2.data)
+        assert adata.samplerate == adata2.samplerate
+
+    @skip_no_mne
+    def test_tldata_to_mne_with_TimeLockData(self):
+        """
+        Test conversion of spy.TimeLockData to mne.EpochsArray.
+
+        This uses epoched data, i.e., data with trial definition and trials of identical length (and offset), i.e., timelocked data.
+        """
+        adata = self.adata
+        assert type(adata) == spy.AnalogData
+        tldata = spy.timelockanalysis(adata, latency="maxperiod")
+        assert type(tldata) == spy.TimeLockData
+        # Convert to MNE EpochsArray
+        epoched = spy.io.mne_conv.tldata_to_mne(tldata)
+        assert type(epoched) == mne.EpochsArray
+
+        # Check dimensions
+        n_times = epoched.get_data().shape[2]
+        assert n_times == self.numSamples
+        n_epochs = epoched.get_data().shape[0]
+        assert n_epochs == self.numTrials
+        n_channels = epoched.get_data().shape[1]
+
+        assert n_times == tldata.trials[0].shape[0]
+        assert n_epochs == len(tldata.trials)
+        assert n_channels == len(tldata.channel)
+
+    @skip_no_mne
+    def test_tldata_to_mne_with_AnalogData(self):
+        """
+        Test conversion of spy.AnalogData that is time locked to mne.EpochsArray.
+
+        This uses epoched data, i.e., data with trial definition and trials of identical length (and offset), i.e., timelocked data.
+        """
+        adata = self.adata
+        assert type(adata) == spy.AnalogData
+        assert adata.is_time_locked == True
+        # Convert to MNE EpochsArray
+        epoched = spy.io.mne_conv.tldata_to_mne(adata)
+        assert epoched.get_data().shape == (self.numTrials, self.numChannels, self.numSamples)
+        for ea in epoched.iter_evoked(): # ea is an mne.EvokedArray
+            assert type(ea) == mne.EvokedArray
+            assert ea.get_data().shape == (self.numChannels, self.numSamples)
+        assert type(epoched) == mne.EpochsArray
+
+        # Check dimensions
+        n_times = epoched.get_data().shape[2]
+        assert n_times == self.numSamples
+        n_epochs = epoched.get_data().shape[0]
+        n_channels = epoched.get_data().shape[1]
+
+        assert n_times == adata.trials[0].shape[0]
+        assert n_epochs == len(adata.trials)
+        assert n_channels == len(adata.channel)
+
+    @skip_no_mne
+    def test_mne_epoched_to_AnalogData(self):
+        """
+        Test conversion of spy.AnalogData that is time locked to mne.EpochsArray.
```

Comment:
Here you probably copy-pasted from the doc string above? It should say `..conversion of mne.EpochsArray to spy.AnalogData` ?!

Reply:
fixed